src/app/api/generateData/orders/route.ts

```diff
@@ -0,0 +1,30 @@
+import { config } from "@/lib/auth";
+import { sampleData } from "@/lib/data";
+import dbConnect from "@/lib/dbConnect";
+import CounterModel from "@/lib/models/CounterModel";
+import OrderModel from "@/lib/models/OrderModel";
+import ProductModel from "@/lib/models/ProductModel";
+import UserModel from "@/lib/models/UserModel";
+import { getServerSession } from "next-auth";
+import { NextRequest, NextResponse } from "next/server";
+
+export const DELETE = async (request: NextRequest) => {
+  const { user } = await getServerSession(config);
+
+  if (!user || !user.isAdmin) {
+    return Response.json(
+      { message: "unauthorized" },
+      {
+        status: 401,
+      }
+    );
+  }
+
+  await dbConnect();
```

Comment:
We need to change this to use a Singleton class, in that way we won't instantiate a new connection per each request. We can do this on a separate PR / ticket.